### ISSUE TITLE
CMake-installation Error

### ISSUE THREAD

**Issue by tod-repo** [jkadbear/gr-lora#2]
I don't want to use Docker installation, I want to use CMake installation, but it always reports errors and prompts CMake error. Can you provide a CMake installation method? Thank you.

Error Msg：
CMake Error at swig/CMakeLists.txt:37 (include):
  include could not find requested file:                          
                                                                  
    GrSwig                                                        
                                                                                                                              
CMake Error at swig/CMakeLists.txt:51 (GR_SWIG_MAKE):
  Unknown CMake command "GR_SWIG_MAKE".

**Comment by jkadbear** [comment]
You can follow the building process in dockerfile：https://github.com/jkadbear/gr-lora/blob/master/Dockerfile. I am not sure if this repo is compatible with the latest version gnuradio.

**Comment by tod-repo** [comment]
I reinstalled version 3.8 of grc and cmake was successful, but when running the case in the examples folder, an error was reported.

Error Msg:

File "/home/r26/Desktop/gr lora/examples/rx_usrp. py", line 144, in__ Init__
Self.lora_ Demod_ 1=lora. demo (8, True, payloadlun, 1, True, False, 25.0, 16, 0, 4, 2)
AttributeError: module 'lora' has no attribute 'demod'

**Comment by tod-repo** [comment]
> You can follow the building process in dockerfile：https://github.com/jkadbear/gr-lora/blob/master/Dockerfile. I am not sure if this repo is compatible with the latest version gnuradio.

I reinstalled version 3.8 of grc and cmake was successful, but when running the case in the examples folder, an error was reported.

Error Msg:

File "/home/r26/Desktop/gr lora/examples/rx_usrp. py", line 144, in__ Init__
Self.lora_ Demod_ 1=lora. demo (8, True, payloadlun, 1, True, False, 25.0, 16, 0, 4, 2)
AttributeError: module 'lora' has no attribute 'demod'

**Comment by jkadbear** [comment]
Did you modified rx_usrp.grc? The original file in this repo uses "lora_demod_0" instead of "lora_Demod_1".

Since this repo shares the same name with other gr-lora projects (e.g., https://github.com/BastilleResearch/gr-lora), there may be some naming problems if you install multiple libs in your environment (the reason why docker is applied). Your error msg shows that the lora module is installed but attribute 'demod' is not found, you can check the actual path and the content of the installed "lora module" for more information.

Besides, try `rx_usrp_collision. py`?

**Comment by tod-repo** [comment]
> Did you modified rx_usrp.grc? The original file in this repo uses "lora_demod_0" instead of "lora_Demod_1".
> 
> Since this repo shares the same name with other gr-lora projects (e.g., https://github.com/BastilleResearch/gr-lora), there may be some naming problems if you install multiple libs in your environment (the reason why docker is applied). Your error msg shows that the lora module is installed but attribute 'demod' is not found, you can check the actual path and the content of the installed "lora module" for more information.
> 
> Besides, try `rx_usrp_collision. py`?

Yes, I have modified the file. My device is not USRP, and for Lora_ I just re dragged the demo to the workspace, but even if I just replaced the usrp Source with the pluto source, the same error still occurred. I can actually use Docker, but in the Docker environment, I also need to install the gr-iio module. When installing gr-iio, I also encountered an Error, which caused me to not know how to handle it.

**Comment by jkadbear** [comment]
`rx_file.grc` and `rx_file_collision.grc` do not depend on any specific RF devices, you can test them in a clean docker environment as a baseline. If you can successfully run them, then the problem maybe device-related.

**Comment by tod-repo** [comment]
> 

I reinstalled an OS for testing today, but there were indeed problems that I found difficult to solve.

**Comment by jkadbear** [comment]
Hope you have solved the problems！
